src/neural/blas/convolution1.cc

```diff
@@ -44,7 +56,7 @@ void Convolution1::Forward(const size_t batch_size, const size_t input_channels,
 
     const float* batch_input = input + i * kSquares * input_channels;
     float* batch_output = output + i * kSquares * output_channels;
-
+#ifndef USE_EIGEN
```

Comment:
Could BLAS and Eigen in theory co-exist?
Then if would be better to have templated functions (e.g. `<bool is_eigen>`) which would be shown up as different backends.

See for example:
cudnn vs cudnn-fp16
https://github.com/LeelaChessZero/lc0/blob/master/src/neural/cuda/network_cudnn.cc#L805
https://github.com/LeelaChessZero/lc0/pull/5#issuecomment-393979584

tensorflow vs tensorflow-cpu
https://github.com/LeelaChessZero/lc0/blob/master/src/neural/network_tf.cc#L342

May be not that straighforward though as different header files are needed (so #ifdefs will be needed in any case), and calling non-included functions in `if(false) {}` won't compile.
Calling them from non-instanciated function template specializations will work though:

```cpp
template<>
DoStuff<false /* is eigen*/>() {
  known_functions();
}

template<>
DoStuff<true /* is eigen*/>() {
  unknown_functions();  // Is fine
}

#if defined(HAVE_EIGEN)
REGISTER_NETWORK("eigen", MakeBlasNetwork<true>, 90)
#endif
#if defined(HAVE_BLAS)
REGISTER_NETWORK("blas", MakeBlasNetwork<false>, 90)
#endif
```

Not sure if easy enough to be bothered with, but that way it would be nicer.

Also there is this brute-force way to allow co-existance:
```cpp
template<bool is_eigen> MyFunction() {
#ifdef HAVE_EIGEN
  if (is_eigen) {
    // Do eigen stuff.
  }
#endif
#ifdef HAVE_BLAS
  if (!is_eigen) {
    // Do blas stuff.
  }
#endif
}

#if defined(HAVE_EIGEN)
REGISTER_NETWORK("eigen", MakeBlasNetwork<true>, 90)
#endif
#if defined(HAVE_BLAS)
REGISTER_NETWORK("blas", MakeBlasNetwork<false>, 90)
#endif
```

Reply:
I don't think there is a reason for this: eigen is only suitable for the build machine, so blas is the way to go for redistributable cpu only binaries. The only use case I can see is for benchmarking between eigen and mkl on a specific machine - is there any other use case?